build_system/ninja_utils.mli

```diff
@@ -0,0 +1,108 @@
+(* This file is part of the Catala build system, a specification language for tax and social
+   benefits computation rules. Copyright (C) 2020 Inria, contributor: Emile Rolley
+   <emile.rolley@tuta.io>
+
+   Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+   in compliance with the License. You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software distributed under the License
+   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+   or implied. See the License for the specific language governing permissions and limitations under
+   the License. *)
+
+(** {1 Ninja expressions} *)
+
+(** Helper module to build ninja expressions. *)
+module Expr : sig
+  (** Represents a ninja expression. Which could be either a literal, a variable references ($_) or
+      a sequence of sub-expressions. *)
+  type t =
+    (* Sequence of sub-expressions. *)
+    | Seq of t list
+    (* Literal string. *)
+    | Lit of string
+    (* Variable reference. *)
+    | Var of string
+
+  val to_string : t -> string
+  (** [to_string exp] returns the string representation of an ninja expression [exp]. *)
+
+  val list_to_string : ?sep:string -> t list -> string
+  (** [list_to_string ?sep ls] returns the string representation of a list [ls] of ninja expressions
+      concatenated with the separator [sep] -- by default ' '. *)
+end
+
+(** {1 Ninja rules} *)
+
+(** Helper module to build ninja rules. *)
+module Rule : sig
+  (* NOTE: is name is really needed despite the use of Map? *)
+  type t = {
+    (* Name of the rule: `rule <name>`. *)
+    name : string;
+    (* Command of the rule: ` command = <command>. *)
+    command : Expr.t;
+    (* Description of the rule: ` description = <description>. *)
+    description : Expr.t option;
+  }
+  (** Represents the minimal ninja rule representation for clerk. *)
+
+  val make : string -> command:Expr.t -> description:Expr.t -> t
+  (** [make name ~command ~description] returns the corresponding ninja rule [Rule.t]. *)
+
+  val to_string : t -> string
+  (** [to_string rule] returns the string representation of the [rule]. *)
+end
+
+(** {1 Ninja builds} *)
+
+(** Helper module to build ninja build statements. *)
+module Build : sig
+  type t = {
+    outputs : Expr.t list;
+    (* NOTE: what's the difference between [Expr.t list] and [Expr.Seq]? => [Expr.Seq] is a unique
```

Comment:
You can put this note in `ocamldoc` format perhaps just under the header "Ninja builds".

Reply:
The note is deprecated, because the string representation of an `Expr.Seq` or a list of `Expr.t` is the same -- one space is added between each sub-expressions.